src/ui/components/Button/index.js

```diff
@@ -35,9 +42,18 @@ export default class Button extends React.Component {
         props.to = to;
       }
 
+      props.className = setClassName({ disabled: props.disabled });
+
+      if (props.disabled) {
+        props.onClick = (event) => {
+          event.preventDefault();
+          log.warn(oneLine`Not calling onClick() for Button link to
+            ${props.href || props.to} because it is disabled`);
+        };
+      }
       return <Link {...props}>{children}</Link>;
     }
 
-    return <button {...props}>{children}</button>;
+    return <button className={setClassName()} {...props}>{children}</button>;
```

Comment:
I'd think `setClassName({ disabled: props.disabled })` down here would work just fine; if `props.disabled` is false-y `classNames` won't output that class. That's easier to follow for me; this approach was a bit confusing to trace.

Reply:
When a `<button>` has the property `disabled={true}` then it already gets styled correctly because of [`:disabled`](https://github.com/mozilla/addons-frontend/blob/master/src/core/css/inc/mixins.scss#L260).